src/transformers/configuration_utils.py

```diff
@@ -463,6 +470,18 @@ def save_pretrained(self, save_directory: Union[str, os.PathLike], push_to_hub:
         if os.path.isfile(save_directory):
             raise AssertionError(f"Provided path ({save_directory}) should be a directory, not a file")
 
+        set_generation_parameters = {}
+        for parameter_name, default_value in GENERATION_DEFAULTS.items():
+            if hasattr(self, parameter_name) and getattr(self, parameter_name) != default_value:
+                set_generation_parameters.update({parameter_name: getattr(self, parameter_name)})
```

Comment:
For my own curiosity - is there a reason for `update` here rather than indexing directly in the dict e.g. 

`set_generation_paramters[parameter_name] = getattr(self, parameter_name)`

?

Reply:
not really, changing to indexing (easier to read)